skll/config.py

```diff
@@ -529,27 +569,27 @@ def _parse_config_file(config_path):
                         "list of objectives")
 
     if not all([objective in SCORERS for objective in grid_objectives]):
-        raise ValueError('Invalid grid objective function/s: {}'
+        raise ValueError('Invalid grid objective function(s): {}'
                          .format(grid_objectives))
 
     # check whether the right things are set for the given task
     if (task == 'evaluate' or task == 'predict') and not test_path:
         raise ValueError('The test set must be set when task is evaluate or '
                          'predict.')
-    if (task == 'cross_validate' or task == 'train') and test_path:
```

Comment:
Because if the user set the `test_path`, may be they wanted to do `evaluate` rather than `cross_validate`  or `train` and forgot to change the task. The SKLL philosophy has always been to make no assumptions and let the user fix the config file lest they run a really long experiment they didn't really want to.